.pre-commit-config.yaml

```diff
@@ -6,3 +6,7 @@ repos:
         args: [--fix, --exit-non-zero-on-fix]
       - id: ruff-format
 
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+    - id: fmt
```

Comment:
I didn't add `cargo-check` since it runs super slow, not suitable for precommit check.